jujugui/static/gui/src/app/state/test-state.js

```diff
@@ -0,0 +1,566 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+describe('State', () => {
+  it('can be instantiated', () => {
+    const state = new window.jujugui.State({});
+    assert.equal(state instanceof window.jujugui.State, true);
+  });
+
+  it('has static ROOT_RESERVED', () => {
```

Comment:
As mentioned, I think these could be module level constants, possibly not even exported, and indirectly tested as part of they logic they apply to. That way, we can also avoid this kind of copy/paste tests.